crates/biome_html_parser/src/syntax/mod.rs

```diff
@@ -81,10 +90,11 @@ fn parse_element(p: &mut HtmlParser) -> ParsedSyntax {
 
     p.bump(T![<]);
     let opening_tag_name = p.cur_text().to_string();
-    let should_be_self_closing = VOID_ELEMENTS
+    let is_component = is_astro_component(&opening_tag_name);
```

Comment:
I think it would be better to have astro specific logic gated behind a parser option.